incubating/argo-cd-sync/argocd_sync.py

```diff
@@ -54,7 +56,27 @@ def main():
     export_variable(CF_OUTPUT_URL_VAR, link_to_app)
 
     ingress_host = get_runtime_ingress_host()
-    execute_argocd_sync(ingress_host)
+
+    # Does the app exist
+    # if not let's wait it has been recorded
+    # but not too long in case of simple misspelling
+    is_app_real=application_exist(ingress_host)
+    count=1
+    while count <3 and is_app_real == False:
+        logging.debug("App does not exist yet %d", count)
+        time.sleep(INTERVAL)
+        count += 1
+        is_app_real=application_exist(ingress_host)
+
+    if application_exist(ingress_host) == False:
+        print(f"ERROR application {APPLICATION} does not seem to exist")
+        sys.exit(3)
+
+    if application_autosync(ingress_host) == False:
```

Comment:
no need to do another request, check autosync while checking existance

Reply:
agreed, I will refactor later as I need another call to check current sync status